### PR TITLE
raftstore: report delta bucket stats

### LINKED ISSUE #12472: Report delta bucket stats
## Development Task

Currently we report full bucket stats in every tick, reporting delta bucket stats seems to be more friendly for PD.

### PR DESCRIPTION
**Pull request by 5kbpers**
### What is changed and how it works?

Issue Number: Close #12472

What's Changed:

```commit-message
Report delta bucket stats
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by lhy1024 — COMMENTED** [review]
LGTM. But it will lose some reports when just transferring leader. cc @bufferflies

File: components/raftstore/src/store/worker/pd.rs [hunk 1]
```diff
@@ -253,6 +253,58 @@ pub struct PeerStat {
...
+
+impl ReportBucket {
+    fn new(current_stat: BucketStat) -> Self {
+        Self {
+            current_stat,
+            ..Default::default()
+        }
+    }
+
+    fn new_report(&mut self, report_ts: UnixSecs) -> BucketStat {
```
Comment (rleungx) [inline components/raftstore/src/store/worker/pd.rs:271]: Do we need a test?

**Comment by bufferflies** [comment]
> LGTM. But it will lose some reports when just transferring leader. cc @bufferflies

After transferring leader, the  bucket state will be recounted again,  PD will smooth the statistics to avoid the hot degree drop quickly.

**Review by rleungx — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Review by gengliqi — APPROVED** [review]

**Pull request merged**